sklearn/ensemble/_hist_gradient_boosting/_predictor.pyx

```diff
@@ -21,46 +21,162 @@ from .common cimport node_struct
 np.import_array()
 
 
+@cython.final
+cdef class TreePredictor:
+    """Tree class used for predictions.
+
+    Parameters
+    ----------
+    nodes_cnt : int
+        Number of nodes in tree.
+    """
+    def __init__(self, int node_cnt):
+        self._nodes.resize(node_cnt)
+
+    cdef node_struct* get(self, int node_idx) nogil:
+        """Get reference to node."""
+        return &self._nodes[node_idx]
+
+    cdef int get_size(self) nogil:
+        """Get number of nodes in tree."""
+        return self._nodes.size()
+
+    @property
+    def nodes(self):
+        """Return nodes as a list of dicts."""
```

Comment:
a list of dicts? I don't understand

Reply:
When accessing a vector in python the struct becomes a dictionary:

```py
%%cython
# distutils: language=c++

from libcpp.vector cimport vector

cdef struct inner:
    int num
        
def hello():
    cdef vector[inner] items
    items.resize(2)
    items[0].num = 4
    items[1].num = 10
    return items
```

```py
hello()
# [{'num': 4}, {'num': 10}]
```

In this case having a public `nodes` attribute is enough, which would return this list of dictionaries.